### PR TITLE
doc: add example interactive aggregation

### PR DESCRIPTION
**Pull request by mattijn**
This PR adds a new example showing interactive aggregation. The example is heavily inspired by [this](https://twitter.com/InsightsMachine/status/1708898945135730727) example by @dwootton.

image

@dwootton, I modified your example a little bit, could you review the introduced changes? I'm not sure if my changes make the goal of the chart more intuitive.

### CODE REVIEW

**Comment by dwootton** [comment]
Thanks for putting this together! 


Love that you also added the constant size scale and filtered out the aggregate points from the view. 

Some nits that don't block the PR but are worth considering: 

1. IMO, the different color breaks the visual connection between aggregated marks and their disaggregated counterparts. returning aggregate marks to the steel blue color could help to convey that these are the same data, but just at different levels of aggregation. 
2. I personally liked the ability to set a float filter value rather than integer steps; however, totally understand that filtering out the aggregate points on the right side of the slider makes these in-between states look weird.

**Comment by mattijn** [comment]
Thanks for the review @dwootton! I think the current approach is in line what you describe. See also [Open the Chart in the Vega Editor](https://vega.github.io/editor/#/url/vega-lite/N4Igxg9gdgZglgcxALlANzgUwO4tJKAFzigFcJSBnAdTgBNCALFAZgAY2AacaYsiygAlMiRoVYcAvpO4AbAIYBPTACcUAbVB15heXhCkVslCDGEADpWQB6a2DpQAdACtKdTLLhoVjqJkLWUOYAttZomAjyALTaupT+lAACaACMjgBMAJyObNax8tbBEBiYlC6U0CAyIMHyKgDW+oSK5pgmYHAqYLJt1ZhQkHQkSKggAB768B50JgCSALIAIgBCAPoASjrDINzEhD1zS8sABJt8SLstbcggAI6k8kRwusThVdyKk1iyMzfrEIRCP1VgAVCC1QgQUobLZQC4gPYHP4AoFQY5giFQyinWHw5qtEz3R57LZvaS7FSPSgwCAqYIaUDwWRAtQ3AAUsVIwXUAHIFisYeceQBdY4APmOTBUpUYEB+AEoqsKZFodHpRoZjDczJYbHYHOV3J5vL5-IEQmEItF8vFCElUhlsrl8oVilgyq5KtVag0mld2p1ur1uP1BttRpQ4AAva6geQIBDSyJA9oUIg7ECUMDyJFacHyEgaLgpABsbGVlwJNyJTxeXmD430ACNC6NamMW1BKCgUmxqlMfocBWdtpXrncHrXSQ3PqNO-p253u8he-3vr8QP9AcCMTosYLRwj-dXJyTXr1qoRKV2aXSGSAmSyTBydFzefy1iO4SLjgAeSWMNKlCygqSoqiA+T6FA8jBOOmDBOYzTvDUdSNKM+LjiopAHNwkCyLSJiJkoyGhhAQxwvolBXhA9SYLQDDMKMaA5qQ1wsNUEzoceE7Es804Zpy9KjJgYzmKyCKATKcozNI0jCtw5h1DBy6aCA0GwSYUpSYO3Cdr8oAkOYpDiDc14IG03Dtj2XA1K2NlUZg5goGwjgpNUzGyKxKAAKySPJIAACRZow8HqqYgK6rY4SRI4CDPIwpBNo4cAQNYwWhZakRRJ4QJhN5rn5WknpQAJaq2suoDwYhs6aH50hAA). Agreed?

**Comment by binste** [comment]
This is really cool! Thank you both :) Just one small comment, should one filter statement include the actual `threshold` value? Else, observations with the exact same IMDB rating as the threshold are dropped.

File: tests/examples_arguments_syntax/interactive_aggregation.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    alt.Chart(source).mark_circle().encode(
+        x=alt.X("IMDB_Rating:Q", bin=alt.Bin(maxbins=10)),
+        y=alt.Y("Rotten_Tomatoes_Rating:Q", bin=alt.Bin(maxbins=10)),
+        size=alt.Size("count():Q", scale=alt.Scale(domain=[0,160]))
+    ).transform_filter(
+        alt.datum["IMDB_Rating"] < threshold
+    ),
+
+    alt.Chart().mark_rule(color="gray").encode(
+        strokeWidth=alt.StrokeWidth(value=6),
```
Comment (joelostblom) [inline tests/examples_arguments_syntax/interactive_aggregation.py:36]: Nice use of the new syntax we just discovered =) We should probably document this somewhere eventually and decide on one to favor in the docs, but that's another issue.

**Review by joelostblom — APPROVED** [review]
This example is really neat, love it!

**Comment by mattijn** [comment]
Thanks for the reviews! Especially thanks to @dwootton for exploring this type of interactive visualizations (and sharing these with the world!👍)
Merging!

**Pull request merged**